android/fastlane/Fastfile

```diff
@@ -0,0 +1,132 @@
+# This file contains the fastlane.tools configuration
+# You can find the documentation at https://docs.fastlane.tools
+#
+# For a list of all available actions, check out
+#
+#     https://docs.fastlane.tools/actions
+#
+# For a list of all available plugins, check out
+#
+#     https://docs.fastlane.tools/plugins/available-plugins
+#
+
+# Uncomment the line if you want fastlane to automatically update itself
+# update_fastlane
+
+default_platform(:android)
+
+platform :android do
+    desc "Create new release"
+    lane :release do |options|
+        ensure_git_status_clean
+        ensure_git_branch( branch: 'main' )
+
+        options_release_number = options[:release_number]
+
+        newVersion = get_version_number(
+            release_number: options_release_number
+        )
+
+        isInteractiveMode = options_release_number == nil
+
+        if !isInteractiveMode
+            UI.message("Release number: #{options_release_number}")
+        else
+            if UI.confirm("Creating release for version=#{newVersion}?\n")
+                if UI.confirm(text:"Are you sure you want to continue?")
+                    do_commit_and_push_tags(newVersion: newVersion)
+                else
+                    UI.error errorMessageCancelled
+                end
+            else
+                UI.error errorMessageCancelled
+            end
+        end
+
+    end
+
+    private_lane :do_commit_and_push_tags do |options|
+        newVersion = options[:newVersion]
+
+        set_version_number(newVersion: "#{newVersion}")
+
+        git_commit(
+            message: "Release #{newVersion}",
+            path: "*",
+            allow_nothing_to_commit: true,
+            skip_git_hooks: true
+        )
+
+        sh "git tag -a #{newVersion} -m '#{newVersion}'"
+        UI.header("#{newVersion} release successfully created. Commit info: #{last_git_commit}")
+
+        if UI.confirm(text:"Do you want to push the tags to the remote?")
```

Comment:
Nit:
- this message makes it seem like you can optionally push the tags or not, but really it's a case of push now vs abort the script. 
- the current text makes it sounds like pushing tags only, but I _think_ it's not just the new tag that will be pushed, but also the commit to main 

Maybe something more like "Are you ready to push the new tag and the updated version number to main" or something would be clearer?

Reply:
That's a good call. I like your suggestion better 🙏